APIs/openEO/Glossary.qmd

```diff
@@ -0,0 +1,78 @@
+
+A **coordinate** represents a specific point in space.
+
+A **feature** is a thing that usually has a geometry (e.g. the outline of an agricultural field, a forest or an urban area) and it may have additional properties assigned (e.g. a name, a color or a population).
+In rare cases features may not have a geometry, which is often referred to as an "empty geometry".
+
+**Geometries** consist of one or more coordinates that may be connected and then form a specific type of geometry, e.g. two points can be connected to a straight line and four straight lines can be connected to rectangle.
+
+Commonly used types of geometries are:
+- Point
+- LineString (connected straight line pieces)
+- Polygon (connected straight line pieces forming a closed ring, possibly with holes - for example a triangle or rectangle)
+
+Multiple geometries of the same type can be combined into a group of geometries, e.g. a so-called "multi point" or  "multi polygon".
+
+Features and geometries are specified by the OGC in the [Simple Feature Access specification](https://www.ogc.org/standards/sfa){target="_blank"} (and ISO 19125). See the specification for more details.
+
+## User-defined function (UDF)
+
+The abbreviation **UDF** stands for [**user-defined function**](https://open-eo.github.io/openeo-python-client/udf.html){target="_blank"}. With this concept, users are able to upload custom code and have it executed e.g. for every pixel of a scene, or applied to a particular dimension or set of dimensions, allowing custom server-side calculations.
+
+## Data Processing modes
+
+Processes can run in three different ways:
+
+1. Results can be pre-computed by creating a ***batch job***.  They are submitted to the back-end's processing system, but will remain inactive until explicitly put into the processing queue. They will run only once and store results after execution. Results can be downloaded. Batch jobs are typically time consuming and user interaction is not possible although log files are generated for them. This is the only mode that allows to get an estimate about time, volume and costs beforehand.
+2. A more dynamic way of processing and accessing data is to create a **secondary web service**. They allow web-based access using different protocols such as [OGC WMS](http://www.opengeospatial.org/standards/wms){target="_blank"} (Open Geospatial Consortium Web Map Service), [OGC WCS](http://www.opengeospatial.org/standards/wcs){target="_blank"} (Web Coverage Service) or [XYZ tiles](https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames){target="_blank"}. These protocols usually allow users to change the viewing extent or level of detail (zoom level). Therefore, computations often run *on demand* so that the requested data is calculated during the request. Back-ends should make sure to cache processed data to avoid additional/high costs and reduce waiting times for the user.
```

Comment:
remove secondary web service, doesn't exist in this context

Reply:
updated